target/linux/ramips/image/mt7620.mk

```diff
@@ -7,16 +7,13 @@ define Build/tplink-header
 		-o $@.new -k $@ -r $(IMAGE_ROOTFS) && mv $@.new $@
 endef
 
-define Build/pad-kernel-ex2700
-	cp $@ $@.tmp && dd if=/dev/zero bs=64 count=1 >> $@.tmp \
-		&& dd if=$@.tmp of=$@.new bs=64k conv=sync && truncate -s -64 $@.new \
-		&& cat ex2700-fakeroot.uImage >> $@.new && rm $@.tmp && mv $@.new $@
-endef
-
-define Build/netgear-header
-	$(STAGING_DIR_HOST)/bin/mkdniimg \
-		$(1) -v OpenWrt -i $@ \
-		-o $@.new && mv $@.new $@
+define Build/netgear-kernel
+	let \
+		bsize="$(subst k,*1024,$(1))" \
+		size="$$(stat -c%s '$@')" \
+		size="((1 + ((64 + size) / bsize)) * bsize) - 64"; \
+	dd if=$@ of=$@.new bs=$$size count=1 conv=sync \
```

Comment:
The second uImage header must be located in the last 64 bytes of the kernel image padded to 64k. The code above would add the uImage header *after* the padding.

This is the relevant code in the bootloader:

```
hdr = (((((ulong)hdr + sizeof(image_header_t) + ntohl(hdr->ih_size))
                / 0x10000) + 1 ) * 0x10000) - sizeof(image_header_t);
if (ntohl(hdr->ih_magic) != IH_MAGIC) {
    puts ("Second part Bad Magic Number\n");
    goto bad_img;
}

```